### PR TITLE
Java API 2.0

### PR DESCRIPTION
**Pull request by likholat**
### Current state
| _C++_ | _Java_ |
|:--------|:----------|
|**ov::Core**|**Core**|
|  read_model(model_path,bin_path) |  read_model(modelPath, weightPath)   |
|  read_model(model_path) |   read_model(modelPath) |
|  compile_model(model, device_name, properties = {}) |  compile_model(model, device) |
|  **ov::Model** |  **Model** |
| get_name  |  get_name |
|  get_batch |  get_batch |
| **ov::CompiledModel**  |  **CompiledModel** |
|  create_infer_request | create_infer_request  |
|  ov::InferRequest |  **InferRequest** |
| infer    | infer  |
| set_input_tensor  | set_input_tensor  |
| get_output_tensor  | get_output_tensor  |
| **ov::Tensor**  | **Tensor**  |
| get_size  |  get_size |
|  Shape get_shape() | int[] get_shape() |
|  T* data() | float[] data() |
| **ov::preprocess::InputInfo**  | **InputInfo** |
| preprocess   | preprocess |
|  tensor  | tensor |
|  model | model |
|  **ov::preprocess::InputModelInfo** | **InputModelInfo** |
| set_layout  | set_layout |
|  **ov::preprocess::PreProcessSteps**  | **PreProcessSteps** |
|  resize |  resize|
| **ov::preprocess::OutputInfo**  | **OutputInfo** |
| tensor  | tensor |
| **ov::preprocess::InputTensorInfo**  |  InputTensorInfo|
|  set_element_type |  set_element_type|
|  set_layout | set_layout |
|  set_spatial_dynamic_shape| set_spatial_dynamic_shape|
|  set_spatial_static_shape| set_spatial_static_shape|

### CODE REVIEW

**Comment by dkurt** [comment]
@allnes, we plan merge this PR first and then add missed methods. Can you check which methods do you need first so we can prioritize?

File: modules/java_api/tests/compatibility/TestsSuite.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package tests.compatibility;
```
Comment (dkurt) [inline modules/java_api/tests/compatibility/TestsSuite.java:1]: Do we need two TestSuite files?

File: modules/java_api/samples/README.md [hunk 2]
```diff
@@ -6,34 +6,40 @@ This guide describes how to run the benchmark applications.
 
 Upon start-up, the application reads command-line parameters and loads a network to the Inference Engine plugin, which is chosen depending on a specified device. The number of infer requests and execution approach depend on the mode defined with the `-api` command-line parameter.
 
-## Build
-Create an environment variable with Inference Engine installation path:
+## Build OpenVINO Java bindings
+Set environment OpenVINO variables:
 ```bash
-export IE_PATH=/path/to/openvino/bin/intel64/Release/lib
+source /openvino/installation/path/setupvars.sh
```
Comment (dkurt) [inline modules/java_api/samples/README.md:12]:
```
source <openvino_install>/setupvars.sh
```

File: modules/java_api/samples/README.md [hunk 3]
```diff
@@ -6,34 +6,40 @@ This guide describes how to run the benchmark applications.
...
 ```
 
-To create java library and java samples for Inference Engine add `-DIE_EXTRA_MODULES` flag in cmake command while building `openvino`:
+Use Gradle to build `java_api.jar` file with OpenVINO Java bindings:
+```bash
+cd /modules/java_api
```
Comment (dkurt) [inline modules/java_api/samples/README.md:17]:
```
cd openvino_contrib/modules/java_api
```

**Comment by dkurt** [comment]
Hi! Please fix CI issues (rebase if needed) and add a note about compatibility package at https://github.com/openvinotoolkit/openvino_contrib/blob/master/modules/java_api/README.md

File: modules/java_api/cpp/compatibility/data.cpp [hunk 4]
```diff
@@ -1,16 +1,16 @@
 #include <inference_engine.hpp>
 
 #include "openvino_java.hpp"
-#include "enum_mapping.hpp"
 #include "jni_common.hpp"
+#include "enum_mapping.hpp"
```
Comment (dkurt) [inline modules/java_api/cpp/compatibility/data.cpp:5]: Please revert this line

File: modules/java_api/cpp/tensor.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                uint8_t *data = (uint8_t *)matDataAddr;
+                *ov_tensor = Tensor(input_type, input_shape, data);
+                break;
+            }
+            case element::f32:
+            {
+                float *data = (float *)matDataAddr;
+                *ov_tensor = Tensor(input_type, input_shape, data);
+                break;
+            }
```
Comment (dkurt) [inline modules/java_api/cpp/tensor.cpp:30]: Add a default case with exception of unexpected data type

File: modules/java_api/org/intel/openvino/Shape.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package org.intel.openvino;
+
+import java.util.Vector;
+
+// public class OVShape extends IEWrapper {
+
+//     protected OVShape(long addr) {
+//         super(addr);
+//     }
```
Comment (dkurt) [inline modules/java_api/org/intel/openvino/Shape.java:9]: Please remove commented code

File: modules/java_api/org/intel/openvino/Tensor.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        super(GetTensorCArray(type.getValue(), dims, cArray));
+    }
+
+    public int get_size() {
+        return GetSize(nativeObj);
+    }
+
+    // public OVShape GetShape() {
+    //     return new OVShape(GetShape(nativeObj));
+    // }
```
Comment (dkurt) [inline modules/java_api/org/intel/openvino/Tensor.java:18]: Commented code

File: modules/java_api/org/intel/openvino/Tensor.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public float[] data() {
+        return asFloat(nativeObj);
+    }
+
+    /*----------------------------------- native methods -----------------------------------*/
+    private static native long GetTensorCArray(int type, int[] shape, long cArray);
+
+    // private static native long GetShape(long addr);
```
Comment (dkurt) [inline modules/java_api/org/intel/openvino/Tensor.java:31]: unused code

File: modules/java_api/tests/TestsSuite.java [hunk 9]
```diff
@@ -1,6 +1,8 @@
+package tests;
+
 import junit.framework.TestSuite;
 
-import org.intel.openvino.*;
+import org.intel.openvino.compatibility.*;
```
Comment (dkurt) [inline modules/java_api/tests/TestsSuite.java:5]: Why new API test uses compatibility package?

File: modules/java_api/tests/CoreTests.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Test
+    public void testLoadNetwork() {
+        Model net = core.read_model(modelXml);
+        CompiledModel compiledModel = core.compile_model(net, device);
+
+        assertTrue(compiledModel instanceof CompiledModel);
+    }
+}
```
Comment (dkurt) [inline modules/java_api/tests/CoreTests.java:44]: I think we need at least one test for every introduced function. Can you please cover the rest?

File: modules/java_api/tests/TestsSuite.java (hunk 9 above)
Comment (likholat) [reply modules/java_api/tests/TestsSuite.java:5]: Extra import

File: modules/java_api/samples/CMakeLists.txt [hunk 11]
```diff
@@ -19,37 +22,42 @@ endif()
...
-            message(WARNING "OPENCV .jar file is disabled or not found, ${IE_SAMPLE_NAME} skipped")
+            message(WARNING "OPENCV .jar file is disabled or not found, ${OV_SAMPLE_NAME} skipped")
             return()
         endif()
     endif()
 
     # Create .jar file from sources
+
+    message(STATUS "!!!!!!!!!!!!!!!!")
+    message(STATUS ${CMAKE_BINARY_DIR} )
```
Comment (dkurt) [inline modules/java_api/samples/CMakeLists.txt:51]: Please remove debug prints

**Review by dkurt — APPROVED** [review]

**Pull request merged**